## Importing libraries and Dataset

In [1]:
import numpy as np
import pandas as pd
agri_df = pd.read_csv("FoodBalanceSheets_E_Africa_NOFLAG.csv",encoding ="ISO-8859-1")


### Viewing Data

In [2]:
agri_df

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

In [3]:
agri_df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

### Using groupby( ) funct. 


In [46]:
agri_df.groupby('Item')['Y2014'].sum()

##thus animal fats = 209460.54

Item
Alcohol, Non-Food        2403.00
Alcoholic Beverages    102410.11
Animal Products         11935.65
Animal fats            209460.54
Apples and products      9499.23
                         ...    
Vegetables, Other      155038.96
Vegetal Products       107145.19
Wheat and products     232670.13
Wine                     4497.36
Yams                   200396.96
Name: Y2014, Length: 119, dtype: float64

In [48]:
agri_df.groupby('Item')['Y2017'].sum()

##thus animal fats = 269617.53

Item
Alcohol, Non-Food        2348.00
Alcoholic Beverages     95581.06
Animal Products         11547.65
Animal fats            269617.53
Apples and products     10198.90
                         ...    
Vegetables, Other      157752.59
Vegetal Products       107655.20
Wheat and products     240047.62
Wine                     4178.02
Yams                   229174.59
Name: Y2017, Length: 119, dtype: float64

In [63]:
np.round((agri_df.iloc[:,8].std()),3)

1603.404

In [64]:
np.round((agri_df.iloc[:,8].mean()),3)

135.236

### Thus mean = 135.236 and std = 1603.404

### Total number and percentage of missing data in 2016 to 2 decimal places

In [72]:
missing_values_2016 = agri_df.loc[:,'Y2016'].isnull().sum()
print(missing_2016)

1535


In [76]:
np.round( ( (missing_values_2016/len(agri_df))*100 ), 2 )

2.52

number of missing data in 2016 = 1535
percentage = 2.52

## Finding correlation

In [79]:
corr=agri_df.corr()
corr

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

In [83]:
corr.iloc[3:,2]

Y2014    0.024457
Y2015    0.023889
Y2016    0.023444
Y2017    0.024254
Y2018    0.024279
Name: Element Code, dtype: float64

Max correlation = 0.024457 of 2014 with 'Element Code'.

In [85]:
agri_df.groupby('Element').sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

###  2017 has the highest sum of Import Quantity of 294559.09

### The total number of the sum of Production in 2014 is 1931287.75

In [92]:
agri_df.groupby(['Element'])['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

In [93]:
agri_df.groupby(['Element'])['Y2018'].sum().max()

2161192.1

### Thus max is Domestic supply quantity

In [96]:
agri_df.groupby(['Element'])['Y2018'].sum().min()

90.0

so the lowest is Tourist consumption(value = 90) then Fat supply quantity (g/capita/day) (value=10258.69) and the 3rd lowest is Protein supply quantity (g/capita/day) with value= 11833.56.

In [101]:
df2 = agri_df.groupby(['Area','Element'],as_index=False)['Y2018'].sum()
df2.iloc[:20,:]

Area                                 Element     Y2018
0   Algeria                Domestic supply quantity  90023.42
1   Algeria                         Export Quantity    811.54
2   Algeria      Fat supply quantity (g/capita/day)    309.09
3   Algeria                                    Feed  10668.00
4   Algeria                                    Food  63736.33
5   Algeria           Food supply (kcal/capita/day)  13288.00
6   Algeria     Food supply quantity (kg/capita/yr)   1509.28
7   Algeria                         Import Quantity  36238.29
8   Algeria                                  Losses   7708.00
9   Algeria                   Other uses (non-food)   3970.08
10  Algeria                              Processing   2479.00
11  Algeria                              Production  60766.67
12  Algeria  Protein supply quantity (g/capita/day)    367.28
13  Algeria                               Residuals    630.00
14  Algeria                                    Seed    825.00
15  Algeria                         Stock Variation   6172.00
16  Algeria           Total Population - Both sexes  42228.00
17   Angola                Domestic supply quantity  59590.22
18   Angola                         Export Quantity   7962.28
19   Angola      Fat supply quantity (g/capita/day)    230.90

So Import Quantity in Algeria is 36238.29

### Finding no. of unique countries in the dataset.

In [30]:
agri_df.groupby(['Area Code'],as_index=False)['Area'].count()

Area Code  Area
0           4  1313
1           7  1424
2          20  1378
3          32  1460
4          35  1260
5          37  1344
6          39  1343
7          45    45
8          46  1413
9          53  1285
10         59  1488
11         62    39
12         72  1095
13         74  1245
14         75  1168
15         81  1354
16         90  1410
17        107  1467
18        114  1560
19        122  1150
20        123  1206
21        129  1439
22        130  1447
23        133  1293
24        136  1195
25        137  1378
26        143  1393
27        144  1427
28        147  1283
29        158  1403
30        159  1474
31        175  1128
32        181  1393
33        184  1426
34        193  1177
35        195  1462
36        196    68
37        197  1242
38        202  1399
39        206    53
40        209  1284
41        215  1419
42        217  1294
43        222  1338
44        226  1458
45        233  1409
46        238  1447
47        251  1451
48        276  1316

Thus we have 48+1=49 unique areas/countries in the dataset.